### ISSUE TITLE
Some losses can't handle large vales in `targets`/`labels`

### ISSUE THREAD

**Issue by v-iashin** [KevinMusgrave/pytorch-metric-learning#744]
Here is an MWE:
```python
from pytorch_metric_learning import losses, miners, testers
import torch
loss_fn = losses.SupConLoss()

N = 100
D = 128

emb = torch.randn(N, D)

for max_target in [10**1, 10**2, 10**3, 10**4, 10**5, 10**6]:
    target = torch.randint(0, max_target, (N,))
    print(f'{target}: {loss_fn(emb, target).item():.4f}')
# 10: 4.9353
# 100: 5.1204
# 1000: 4.9873
# 10000: 4.7149
# 100000: 0.0000 - hm?
# 1000000: 0.0000
```

I am using `pytorch-metric-learning   2.8.1              pyh101cb37_1    conda-forge`

**Comment by KevinMusgrave** [comment]
I think [SupConLoss](https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#supconloss) requires at least 1 positive pair in the batch. 

So maybe what's happening is that as you increase `max_target`, it becomes more and more unlikely for there to be a positive pair in the batch (if you're just randomly sampling). If there are no positive pairs, it skips the loss computation and returns 0, as you can see here: https://github.com/KevinMusgrave/pytorch-metric-learning/blob/216a792c22b0154d3d3bf008a7317edc7d677e71/src/pytorch_metric_learning/losses/supcon_loss.py#L15-L39

**Comment by v-iashin** [comment]
ok, fair enough
